### PR TITLE
SearchServiceTests.testParseSourceValidation failure

### PR DESCRIPTION
**Pull request by drempapis**
This test has been failing due to #113081, which deleted the incoming search validation on the data node for bwc. 

In this work, #105150, validation has been moved to the coordinate nodes. The following is a matrix of the validation Exception messages of the `SearchService` moving to `SearchRequest`, including a test creation checking.

```
SearchService:validateSearchSource                                                   SearchRequest:validate        SearchRequestTest::testValidate
 
	"disabling [track_total_hits] is not allowed in a scroll context. Yes yes
	`search_after` cannot be used in a scroll context. 				    yes                            yes
	`from` parameter must be set to 0 when `search_after` is used.                      yes                            yes
	cannot use `collapse` in a scroll context                                           yes                            yes
	cannot use `collapse` in conjunction with `rescore`                                 yes                            yes
	[slice] can only be used with [scroll] or [point-in-time] requests                  yes                            yes 
	[stored_fields] cannot be disabled if [_source] is requested                        yes                            yes
	[stored_fields] cannot be disabled when using the [fields] option"                  yes                            yes

```

### CODE REVIEW

**Review by javanna — APPROVED** [review]
LGTM thanks!

File: server/src/test/java/org/elasticsearch/search/SearchServiceTests.java [hunk 1]
```diff
@@ -2930,78 +2928,6 @@ public void testSlicingBehaviourForParallelCollection() throws Exception {
...
-            searchRequest.source().fetchField("field");
-            assertCreateContextValidation(
-                searchRequest,
-                "[stored_fields] cannot be disabled when using the [fields] option",
-                indexService,
-                service
-            );
-        }
-    }
-
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/search/SearchServiceTests.java:3004]: I think that you can delete the assertCreateContextValidation method too

**Comment by javanna** [comment]
Don't forget to close #115936 once this gets merged ;)

File: server/src/test/java/org/elasticsearch/search/SearchServiceTests.java (hunk 1 above)
Comment (drempapis) [reply server/src/test/java/org/elasticsearch/search/SearchServiceTests.java:3004]: Done!

**Pull request merged**